### ISSUE TITLE
CUDA out-of-memory while render_factor=15 with an 8GB VRAM GTX1070 (Windows 10)

### ISSUE THREAD

**Issue by lf-openthos** [jantic/DeOldify#89]
The testing code is here:
>import os
>from fasterai.visualize import *
>plt.style.use('dark_background')
>render_factor=15
>vis = get_artistic_image_colorizer(root_folder=Path("E:/hpc-demo-video/DeOldify"),render_factor=render_factor)
>vis.plot_transformed_image("test_images/poolparty.jpg", render_factor=render_factor)
>vis.plot_transformed_image("test_images/1852GatekeepersWindsor.jpg", render_factor=render_factor)
>vis.plot_transformed_image("test_images/1850SchoolForGirls.jpg", render_factor=render_factor)

I'm using render_factor of 15, but I still get a OOM error

>RuntimeError: CUDA out of memory. Tried to allocate 2.05 GiB (GPU 0; 8.00 GiB total capacity; 432.71 MiB already allocated; 5.79 GiB free; 77.29 MiB cached)

**Comment by jantic** [comment]
Which operating system?

**Comment by lf-openthos** [comment]
Windows 10 build 1903
i7-8700K with 16GB RAM
Nvidia gtx1070 driver 417.71
cuda 10 / cudnn 7

**Comment by jantic** [comment]
I'll keep this issue open for now but Windows is basically just not a priority right now.  This was an issue reported with the previous version of DeOldify so it's familiar.

**Comment by dan64** [comment]
I own a GTX 1070 with 8GB and I had the same problem. To understand the problem I installed in a dedicated partition Ubuntu 19.04, and I was able to run the notebooks only using the Docker : max memory usage 7.7GB. Looking to Anaconda repo, I discovered the were available older version of pytorch, so I switched back to Windows 10 and using Anaconda I installed these old versions with the command:
       `conda install pytorch=1.1.0 torchvision=0.3.0 cudatoolkit=10.0 -c pytorch`
With these versions I was able to run the notebooks in Windows 10. 
**Now the strange thing**:  In Ubunto the max memory usage (render_factor=25) was about 7.7GB, in Windows 10 only 4GB. The amount of RAM necessary to run the scripts in Windows is much lower respect to Linux (Ubuntu 19.04). I suspect that the nvidia drivers in Windows are able to manage the GPU memory with more efficiency even if the encoding speed is almost the same. I set **render_factor=64** and I was able to colorize a video in HD using only 4.5GB in Windows 10. After this discovery I will stop using Linux for my Deep Learning tests and I will move to Windows 10.
